include/ginkgo/core/log/stream.hpp

```diff
@@ -206,16 +206,15 @@ class Stream : public Logger {
      *                 includes always printing residuals and other information
      *                 which can give a large output.
      */
-    explicit Stream(
-        std::shared_ptr<const gko::Executor> exec,
-        const Logger::mask_type& enabled_events = Logger::all_events_mask,
-        std::ostream& os = std::cout, bool verbose = false)
-        : Logger(exec, enabled_events), os_(os), verbose_(verbose)
+    explicit Stream(std::shared_ptr<const gko::Executor> exec,
+                    const Logger::mask_type& enabled_events, std::ostream& os,
+                    bool verbose)
+        : Logger(exec, enabled_events), os_(&os), verbose_(verbose)
     {}
 
 
 private:
-    std::ostream& os_;
+    std::ostream* os_;
```

Comment:
is any consideration here?

Reply:
yes, reference members are a bad idea in general, because every assignment to the object gets passed through to the member, since the reference can't be reassigned.